### PR TITLE
chore: Add rector cache to workflows and template rector.php

### PR DESCRIPTION
**Pull request by samsonasik**
**Description**

Per @kenjis request at https://github.com/codeigniter4/CodeIgniter4/pull/8655#issuecomment-2016616860

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [ ] Conforms to style guide

### CODE REVIEW

**Comment by samsonasik** [comment]
Ready to merge 👍

File: rector.php [hunk 1]
```diff
@@ -50,6 +51,12 @@
 
     $rectorConfig->parallel();
 
+    // Github action cache
+    $rectorConfig->cacheClass(FileCacheStorage::class);
+    if (is_dir('/tmp')) {
```
Comment (MGatner) [inline rector.php:56]: Should this be `/tmp/rector`? Or will it create on demand?

Comment (samsonasik) [reply rector.php:56]: that `/rector` will be created when not exists, but `/tmp` path may not be exists in some OS, eg: windows

https://github.com/rectorphp/rector-src/blob/25f3af626f841f7e62606a3a9aa64789a460f74c/src/Caching/CacheFactory.php#L35-L37

of course, the better is make is ci detected verify part of RectorConfig API.

https://github.com/rectorphp/rector-src/blob/25f3af626f841f7e62606a3a9aa64789a460f74c/config/config.php#L32

but currently prefixed and used internally.

**Review by MGatner — APPROVED** [review]

**Comment by MGatner** [comment]
Looks good!

**Pull request merged**